perceval/utils/matrix.py

```diff
@@ -145,19 +146,44 @@ def random_unitary(n: int, parameters: Optional[Union[np.ndarray,list]] = None)
         :return: a numeric Matrix
         """
         if parameters is not None:
-            assert len(parameters) == 2*n**2, "parameters has not the right size: should be %d, and is %d" % (
-                2*n**2, len(parameters)
-            )
-            a = np.reshape(parameters[:n**2], (n, n))
-            b = np.reshape(parameters[n**2:], (n, n))
+            warnings.warn("use parametrized_unitary(parameters) instead to create a parametrized unitary "
+                          "matrix, version=0.11", DeprecationWarning)
+            assert len(parameters) == 2 * n ** 2, "parameters do not have the right size: should be %d, and is %d" % (
+                2 * n ** 2, len(parameters))
+            a = np.reshape(parameters[:n ** 2], (n, n))
+            b = np.reshape(parameters[n ** 2:], (n, n))
             u = a + 1j * b
+            (q, r) = np.linalg.qr(u)
+            r_diag = np.sign(np.diagonal(np.real(r)))
+            n_u = np.zeros((n, n))
+            np.fill_diagonal(n_u, val=r_diag)
+            return MatrixN(np.matmul(q, n_u))
         else:
             u = np.random.randn(n, n) + 1j*np.random.randn(n, n)
+            (q, r) = np.linalg.qr(u)
+            r_diag = np.sign(np.diagonal(np.real(r)))
+            n_u = np.zeros((n, n))
+            np.fill_diagonal(n_u, val=r_diag)
+            return Matrix(np.matmul(q, n_u))
+
+    @staticmethod
+    def parametrized_unitary(n: int, parameters: Union[np.ndarray,list] = None) -> MatrixN:
```

Comment:
Can't we calculate n from the len of parameters ? And parameters can't be None right (default value) ?
So the method signature would be `def parametrized_unitary(parameters: Union[np.ndarray,list])`

Reply:
Yes, we can calculate n from the length of parameters however, it would involve a square root and I usually not prefer to have that as it would be a float and then you make it into an integer. Furthermore, with this signature the user can easily see that they are asking for a size 'n' matrix with '2*n^2' parameters.

Yes, in this case having `None` makes no sense. I will remove it. Thanks :)